siad/main.go

```diff
@@ -111,8 +113,10 @@ func startEnvironment(*cobra.Command, []string) {
 	}
 
 	// join the network
+	//
+	// TODO: This should proabably happen in newDaemon.
 	if !config.Siacore.NoBootstrap {
-		go d.bootstrap()
+		go d.gateway.Bootstrap(modules.BootstrapPeers[0])
```

Comment:
d.listen isn't called in New?

I definitely think that New should return something that's fully initialized and running. I also think that `NoBootstrap` belongs in `daemonConfig`, in the gateway config section.